datanode/sqlstore/cursor.go

```diff
@@ -39,12 +39,14 @@ const (
 )
 
 type ColumnOrdering struct {
-	Name    string
-	Sorting Sorting
+	Name         string
+	Sorting      Sorting
+	CursorColumn bool
+	Prefix       string
```

Comment:
This is starting to get a bit muddled again, I'm not sure `CursorColumn` and `Prefix` are belong in something called `ColumnOrdering`.. And I'm not exactly sure what they're for :)

Reply:
The do belong on the ColumnOrdering, I will add documentation to the struct, but basically:

CursorColumn is a flag that indicates the Column should be included as part of the Cursor ordering constructs, without it, you are automatically assuming that each column listed should be part of the cursor sorting, and that's not always the case which was why I needed to add it.

Prefix was added because in some complex queries like the one for aggregated balances where you have a more complex query involving more than one table and you need to specify which table a particular column comes from, then if prefix is defined, it will be pre-pended to the column specified. 

I did try to do it by using the full <prefix>.<column> as the column name without adding the prefix to the ColumnOrdering struct, but that causes other problems with matching column names to struct fields when you are extracting the results.